Library/Formula/opencoarrays.rb

```diff
@@ -0,0 +1,85 @@
+class Opencoarrays < Formula
+  desc "open-source coarray Fortran ABI, API, and compiler wrapper"
+  homepage "http://opencoarrays.org"
+  url "https://github.com/sourceryinstitute/opencoarrays/releases/download/1.2.1/opencoarrays-1.2.1.tar.gz"
+  sha256 "2771a8f6d44f6492d0b30731ab7b7803bfea1a14aab4f6936ddd2a400956e2bb"
+
+  head "https://github.com/sourceryinstitute/opencoarrays.git"
+
+  option "without-test", "Skip build time tests (requires properly configured MPICH and firewall settings)"
+
+  depends_on :fortran
+  depends_on :mpi => [:cc, :f90]
+  depends_on "cmake" => :build
+
+  # GCC >= 5.2 required; no other Homebrew supported compilers are compatible
+  # but the mpicc compiler may be clang (which is ABI compatible with GCC)
+  fails_with :llvm
+  fails_with :gcc
+  fails_with :gcc_4_0
+  ("4.3".."4.9").each do |series|
+    fails_with :gcc => series
+  end
+  fails_with :gcc => "5.1"
+
+  def install
+    mkdir "build" do
+      ENV["FC"] = "mpif90"
+      ENV["CC"] = "mpicc"
```

Comment:
> […], but I don't have enough experience building MPI packages with CMake to know what is "the right way to do it"™

Neither do I, so I can't tell you what the right way is in this constellation, but only what would be desirable given how other build systems deal with MPI. We're not asking for an immediate and all-encompassing solution, but if upstream is aware of the issue and open to your future contribution, this will benefit everyone who tries to package this piece of software; and it will lower everyone's maintenance burden and need for ugly workarounds.